src/main/java/net/hypixel/nerdbot/generator/StringColorParser.java

```diff
@@ -9,310 +8,310 @@
+                        this.setItalic(true);
+                        charIndex = closingIndex + 2;
+                        continue;
+                    }
 
-                        // Special case for a specialSubString
-                        if (description.charAt(i) == ':') {
-                            specialSubStringFlag = true;
-                            specialSubStringIndex = i;
-                            continue;
-                        }
+                    // checking if the command is a color code
+                    MCColor color = (MCColor) findValue(colors, selectedCommand);
+                    if (color != null) {
+                        this.setColor(color);
+                        charIndex = closingIndex + 2;
+                        continue;
+                    }
 
-                        if (i == 99) {
-                            endCharIndex = -1;
-                            break;
-                        }
+                    // checking if the command is a gemstone type
+                    Gemstone gemstone = (Gemstone) findValue(gemstones, selectedCommand);
+                    if (gemstone != null) {
+                        // replacing the selected space with the stat's text
+                        String replacementText = "%%DARK_GRAY%%" + gemstone.getId() + "%%GRAY%%";
+                        description.replace(charIndex, closingIndex + 2, replacementText);
+                        continue;
+                    }
 
-                        if (i + 2 > description.length()) {
-                            endCharIndex = -1;
-                            break;
-                        }
+                    String extraData = "";
+                    // checking if the command has extra special flags
+                    if (selectedCommand.indexOf(':') != -1) {
+                        int specialFlagIndex = selectedCommand.indexOf(':');
+                        extraData = selectedCommand.substring(specialFlagIndex + 1);
+                        selectedCommand = selectedCommand.substring(0, specialFlagIndex);
+                    }
+                    // checking if the command is a stat
+                    Stat stat = (Stat) findValue(stats, selectedCommand);
+                    if (stat != null) {
+                        // replacing the selected space with the stat's text
+                        String replacementText = "%%" + stat.getColor() + "%%" + extraData + stat.getId() + "%%GRAY%%";
+                        description.replace(charIndex, closingIndex + 2, replacementText);
+                        continue;
                     }
 
-                    // If we can't find the end percents, just continue
-                    if (endCharIndex != -1) {
-                        // Move away from color code
+                    // creating an error message showing the available stats, gemstones and color codes available
```

Comment:
selectedCommand should strip `[^a-zA-Z0-9_ ]` before this statement to prevent weird string injections